core/schema/git.go

```diff
@@ -167,6 +170,51 @@ func (s *gitSchema) tag(ctx context.Context, parent *core.GitRepository, args ta
 	}, nil
 }
 
+type tagsArgs struct {
+	Patterns dagql.Optional[dagql.ArrayInput[dagql.String]] `name:"patterns"`
+}
+
+func (s *gitSchema) tags(ctx context.Context, parent *core.GitRepository, args tagsArgs) ([]string, error) {
```

Comment:
Can we consolidate the implementation of this with https://github.com/dagger/dagger/blob/3dfbb7db81aa82db37a11d92cf4fa41239e173f0/core/schema/git.go#L254-L255

Reply:
Not sure to follow, do you mean adding the `--symref` flag ? Otherwise, this seems to be a pretty equivalent implementation apart the fact that we rely on the `git.URL` instead of having it as an argument ; and the fact that we can add some glob patterns ?

Reply:
Yeah, this is the same implementation - so we should avoid having two copies of the same implementation, and instead have one function for all of this.

Otherwise, this can introduces bugs and other confusions if the two implementations diverge over time.

Reply:
Sure yes, the purpose is indeed to replace it with this call 🙏 I am more stuck at the moment on how to mount the auth socket to have an authenticated call instead of relying on the underlying os.exec